File: flutter/lib/views/colors.dart
import 'package:flutter/material.dart';

class AppColors {
    static int primary = 0xff037afb;
    static Color backgroundColor = new Color(0xfff4f4f4);
    static Color borderColor = new Color(0xffcacaca);
    static Color textColor = new Color(0xff212529);
    static Color textFieldBorderColor = new Color(0xffced4da);
    static Color textFieldColor = new Color(0xff495057);
    static Color primaryColor = new Color(primary);
    static Color primaryTextColor = new Color(primary);
    static Color buttonColor = new Color(0xffffffff);
    static Color sectionBackgroundColor = new Color(0xfffefefe);
    static Color errorBackgroundColor = new Color(0xfff8d7da);
    static Color errorTextColor = new Color(0xff721c24);
    static Color errorBorderColor = new Color(0xfff5c6cb);
    static Color warningBackgroundColor = new Color(0xfffff3cd);
    static Color warningTextColor = new Color(0xff856404);
    static Color warningBorderColor = new Color(0xffffeeba);

    static MaterialColor primaryMaterialColor = MaterialColor(primary, AppColors.colorMap());

    static Map<int, Color> colorMap() {
      return {
         50: new Color(0xffe3f3fe),
        100: new Color(0xffbbe1fe),
        200: new Color(0xff8fceff),
        300: new Color(0xff61bbfe),
        400: new Color(0xff3aacff),
        500: new Color(primary),
        600: new Color(0xff008ff1),
        700: new Color(0xff007dde),
        800: new Color(0xff006bcc),
        900: new Color(0xff004dad),
      };
    }
}

File: ios/socialnetwork/views/Colors.swift
//
//  Colors.swift
//  socialnetwork
//
//  Created by Alan Quintiliano on 21/01/20.
//  Copyright © 2020 Alan Donizete Quintiliano. All rights reserved.
//

import UIKit
import SwiftUI

class Colors {
    static let backgroundColor = UIColor(argb: 0xfff4f4f4);
    static let borderColor = UIColor(argb: 0xffcacaca);
    static let textColor = UIColor(argb: 0xff212529);
    static let textFieldBorderColor = UIColor(argb: 0xffced4da);
    static let textFieldColor = UIColor(argb: 0xff495057);
    static let primaryColor = UIColor(argb: 0xff037afb);
    static let primaryTextColor = UIColor(argb: 0xff037afb);
    static let buttonColor = UIColor(argb: 0xffffffff);
    static let sectionBackgroundColor = UIColor(argb: 0xfffefefe);
    static let errorBackgroundColor = UIColor(argb: 0xfff8d7da);
    static let errorTextColor = UIColor(argb: 0xff721c24);
    static let errorBorderColor = UIColor(argb: 0xfff5c6cb);
    static let warningBackgroundColor = UIColor(argb: 0xfffff3cd);
    static let warningTextColor = UIColor(argb: 0xff856404);
    static let warningBorderColor = UIColor(argb: 0xffffeeba);

    static func toColor(color: UIColor) {
        
    }
}

extension UIColor {
    convenience init(red: Int, green: Int, blue: Int, a: Int = 255) {
        self.init(
            red: CGFloat(red) / 255.0,
            green: CGFloat(green) / 255.0,
            blue: CGFloat(blue) / 255.0,
            alpha: CGFloat(a) / 255.0
        )
    }

    convenience init(rgb: Int) {
        self.init(
            red: (rgb >> 16) & 0xFF,
            green: (rgb >> 8) & 0xFF,
            blue: rgb & 0xFF
        )
    }

    convenience init(argb: Int) {
        self.init(
            red: (argb >> 16) & 0xFF,
            green: (argb >> 8) & 0xFF,
            blue: argb & 0xFF,
            a: (argb >> 24) & 0xFF
        )
    }

    func toColor() -> Color {
        return Color(self)
    }
}
